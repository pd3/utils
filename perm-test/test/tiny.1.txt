

perm-test -t test/tiny.1.tgt.txt -b test/tiny.1.bg.txt -f test/tiny.1.fai -c test/tiny.1.calls.txt -n 1e9
perm-test -t test/tiny.1.tgt.bed -b test/tiny.1.bg.bed -f test/tiny.1.fai -c test/tiny.1.calls.bed -n 1e9

    123456789012345678901
        bbbbb  tt  bbb
               cccc
     .... .... oo.. ....
      .... .... o... ....
       .... ...o ....
        .... ..oo ....
         .... .oo. ....
    
    exact: 5./17 = 0.29411764705882354
    perm-test:     0.294090135
    err: 2.7512e-05


perm-test -t test/tiny.1.tgt.txt -b test/tiny.1.bg.txt -f test/tiny.1.fai -c test/tiny.1.calls.txt -n 1e9 --no-bg-overlap

    123456789012345678901
        bbbbb  tt  bbb
               cccc
     .... .... oo.. ....
      .... .... x... ....
       .... ...x ....
        .... ..oo ....
         .... .oo. ....
    
    exact: 3./17 = 0.17647058823529413
    perm-test:     0.1764850
    err: 1.441e-05


perm-test -t test/tiny.1.tgt.txt -b test/tiny.1.bg.txt -f test/tiny.1.fai -c test/tiny.1b.calls.txt -n 1e9
perm-test -t test/tiny.1.tgt.txt -b test/tiny.1.bg.txt -f test/tiny.1.fai -c test/tiny.1b.calls.txt -n 1e9 --no-bg-overlap

    123456789012345678901
        bbbbb  tt  bbb
               c
        .....  oo  ...
    
    exact: 2./10 = 0.2
    perm-test:     0.2
    err: 0



perm-test -t test/tiny.1.tgt.txt -b test/tiny.1.bg.txt -f test/tiny.1.fai -c test/tiny.1c.calls.txt -n 1e9

    123456789012345678901
        bbbbb  tt  bbb
               cccc
               c
    
    exact: 5./17 * 2./10 = 0.058823529411764705
    perm-test:             5.999062e-02
    err: 5.27e-06

perm-test -t test/tiny.1.tgt.txt -b test/tiny.1.bg.txt -f test/tiny.1.fai -c test/tiny.1c.calls.txt -n 1e9 --no-bg-overlap

    exact: 3./17 * 2./10 = 0.03529411764705882
    perm-test:             0.03529223
    err: 1.8876e-06


