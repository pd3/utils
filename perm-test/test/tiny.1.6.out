# NITER_ROUNDS:
#    - number of iterations total
#    - number of batches
# NCALLS:
#    - number of calls used
#    - number of calls skipped because of their size or inaccessibility
# TEST_ENR:
#    - number of iterations
#    - number of times the simulation had the same or more hits than observed in input data
#    - P-value derived from the two fields
# TEST_DPL:
#    - number of iterations
#    - number of times the simulation had the same or fewer hits than observed in input data
#    - P-value derived from the two fields
# TEST_FOLD:
#    - number of hits in the input data
#    - average number of hits in simulations
#    - average of standard deviations approximated from each batch using the current average estimate
# DIST, the distribution of number of hits in the simulations:
#    - numbef of hits in a simulation: the start of the half-closed, half-open distribution bin [beg,end)
#    - numbef of hits in a simulation: the end of the half-closed, half-open distributin bin [beg,end)
#    - number of simulations with this many hits (raw count)
#    - number of simulations with this many hits (density)
SEED	1536663761
NITER_ROUNDS	1.000000e+05	1
MSG	Maximum chromosome randomization error due to length discretization: 0.0e+00%
NCALLS	2	0
TEST_ENR	100000	3543	3.543000e-02
TEST_DPL	100000	100000	1.000000e+00
TEST_FOLD	2	0.376980	0.552925
DIST	0	1	65845	6.584500e+04
DIST	1	2	30612	3.061200e+04
DIST	2	3	3543	3.543000e+03
